tests/functional/keystoneapi_controller_test.go

```diff
@@ -149,9 +150,15 @@ var _ = Describe("Keystone controller", func() {
 			th.ExpectCondition(
 				keystoneApiName,
 				ConditionGetterFunc(KeystoneConditionGetter),
-				condition.MemcachedReadyCondition,
+				keystonev1.KeystoneRabbitMQTransportURLReadyCondition,
 				corev1.ConditionFalse,
 			)
+			th.ExpectCondition(
+				keystoneApiName,
+				ConditionGetterFunc(KeystoneConditionGetter),
+				condition.MemcachedReadyCondition,
+				corev1.ConditionUnknown,
```

Comment:
It looks like `condition.MemcachedReadyCondition` used to be `corev1.ConditionFalse`.  Is there a particular reason it has been changed to `corev1.ConditionUnknown`?

Reply:
This is due to the order of execution. Because the setting of the condition.MemcachedReadyCondition is after KeystoneRabbitMQTransportURLReadyCondition, it never gets set, because it will exit earlier with an error

See the line where MemcachedReadyCondition is set and earlier lines here: https://github.com/openstack-k8s-operators/keystone-operator/pull/325/files/4ed509ca5748368853561e6f547235ca2bdc2701#diff-89511165055af64058394ac61e20f832806f3f325b9cb3771f723d7e6cb94c8cR709

Mauricio will add a test to execute this portion.

Reply:
Ack, thank you for the explanation